### PR TITLE
prototype: status for comments

### LINKED ISSUE #2589: prototype: status for comments
https://trello.com/c/QGPolFbe/186-status-zu-kommentaren-einblenden

we opted for a frontend-only prototype approach here

### PR DESCRIPTION
**Pull request by Entkenntnis**
a lot to be done here, but hey: it persists the state

https://frontend-git-2589-prototype-status-for-comments-serlo.vercel.app/discussions

remaining points:
- [x] side quest: add 4px padding-y
- [x] add sa events for measurement
- [x] limit access (moderators + creator of thread)
- [x] styling
- [x] clean up and brush up
- [ ] ~show status within entity (-> pending)~

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -44,4 +44,9 @@ model MitmachWoche {
   isProduction Boolean
   event        String   @db.VarChar(24)
   timestamp    DateTime @default(now())
+}
+
+model PrototypeThreadStatus {
+  threadId     String @db.VarChar(24) @id
+  status       String @db.VarChar(24)
```
Comment (elbotho) [inline prisma/schema.prisma:51]: easiest db migration ever 👍

Comment (Entkenntnis) [reply prisma/schema.prisma:51]: creating a table is nothing scary anymore :)

File: src/pages/api/frontend/thread-status/get/[id].ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    const result = await prisma.prototypeThreadStatus.findFirst({
+      where: { threadId: id as string },
+    })
+    res.send(result ? result.status : 'none')
+    return
+  } catch (e) {
+    // eslint-disable-next-line no-console
+    // console.log(e)
+  }
+  res.send('bad')
```
Comment (elbotho) [inline src/pages/api/frontend/thread-status/get/[id].ts]: Maybe at least `res.status(400).send('bad')` :)

File: src/pages/api/frontend/thread-status/get/[id].ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import type { NextApiRequest, NextApiResponse } from 'next'
+
+import { prisma } from '@/helper/prisma'
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  try {
+    const { id } = req.query
```
Comment (elbotho) [inline src/pages/api/frontend/thread-status/get/[id].ts:10]: more out of curiosity: does Prisma do any sanitation/validation of the inputs?

**Review by elbotho — APPROVED** [review]
added two comments, but can be tested 👍

File: src/pages/api/frontend/thread-status/get/[id].ts (hunk 3 above)
Comment (Entkenntnis) [reply src/pages/api/frontend/thread-status/get/[id].ts:10]: normal sql validations apply (unique value, data type)

**Pull request merged**